Spektrum SRXL2 implementation

MiguelFAlvarez opened the pull request:
This PR includes support for Spektrum SRXL2. This is based off the betaflight PR https://github.com/betaflight/betaflight/pull/8598 but there are a lot more changes here as some of the other spektrum features required did not yet exist in inav.

As opposed to the betaflight PR which uses the UART Idle interrupt, this code polls the idle flag in the rcFrameStatusFn as requested.

This PR also depends on https://github.com/iNavFlight/inav/pull/5757 which at this time has been approved, but not yet merged. Until it is merged, compile checks won't pass.

digitalentity commented on src/main/drivers/vtx_common.h:
```diff
@@ -62,6 +62,39 @@ typedef enum {
+#define VTX_COMMON_BAND_RACE      5
+
+// RTC6705 RF Power index 25 or 200 mW
+#define VTX_6705_POWER_25      1
```
6705 is not supported on INAV

MiguelFAlvarez replied:
I removed all of the vtx code with the exception of a few placeholders blocked by `USE_SPEKTRUM_VTX_CONTROL`. If anything, will just make a pull request later on with support for this.

digitalentity commented on src/main/drivers/vtx_common.h:
```diff
@@ -62,6 +62,39 @@ typedef enum {
+#define VTX_6705_POWER_25      1
+#define VTX_6705_POWER_200     2
+
+// SmartAudio "---", 25, 200, 500, 800 mW
```
We have unified frequency/power control now and  number of power levels is dependent on the driver and actual VTX used. Hard coded values are not necessary.

digitalentity commented on src/main/fc/cli.c:
```diff
@@ -2624,6 +2625,59 @@ static void cliEleresBind(char *cmdline)
+        default:
+            cliPrint("Not supported.");
+            break;
+#if defined(USE_RX_FRSKY_SPI)
```
Not supported

MiguelFAlvarez replied:
Removed

digitalentity commented on src/main/io/spektrum_vtx_control.c:
```diff
@@ -0,0 +1,233 @@
+
+// RF Power Index translation tables. No generic power API available.....
+
+#ifdef USE_VTX_TRAMP
```
Please don't use VTX-protocol specific tables. Use unified power level API and note that power level is dependent on the driver and actual hardware connected.

digitalentity commented on src/main/io/spektrum_vtx_control.c:
```diff
@@ -0,0 +1,233 @@
+        }
+#endif
+        // Seems to be no unified internal VTX API standard for power levels/indexes, VTX device brand specific.
+        const uint8_t power = convertSpektrumVtxPowerIndex(vtx.power);
```
We don't have a fixed mapping between power level and actual power. It's also driver-independent. Please don't use any hardcoded Smartaudio/Tramp/whatever conversions. Use run-time conversion if possible instead so it will actually match the capabilities of the VTX connected.

digitalentity reviewed (changes requested):
Looks like this PR still uses RTC6705 which we don't support anymore and doesn't account for FFPV 2.4 GHz VTX.

To avoid creating more dependencies, please don't use any VTX-specific tables for power or frequency. Use generic power indexes (but note that they are VTX specific and actual power/index mapping is determined at run-time to match capabilities of the actual hardware used).

What we could consider is to add a protocol-agnostic API to convert from mW of power to VTX power index so we can do all the conversions within the VTX driver and keep control code nice and simple.

digitalentity commented on src/main/build/version.h:
```diff
@@ -16,7 +16,7 @@
 
 #define FC_VERSION_MAJOR            2  // increment when a major release is made (big new feature, etc)
-#define FC_VERSION_MINOR            6  // increment when a minor release is made (small new feature, change etc)
+#define FC_VERSION_MINOR            5  // increment when a minor release is made (small new feature, change etc)
```
Minor nit

MiguelFAlvarez replied:
Oops, didn't mean to commit that lol. Just did that so I could use the configurator.

MiguelFAlvarez replied:
Fixed and squashed

digitalentity reviewed (changes requested):
Minor nit, otherwise LGTM

MiguelFAlvarez committed:
Added Spektrum SRXL2 support.
This consists of various files/changes brought over from betaflight with modifications to operate in the current state of inav.
It also includes files/changes that were not a part of the betaflight SRXL2 merge, as the previous bidi srxl implementation was not yet implemented either, and SRXL2 has some dependencies on the Spektrum telemetry structuring from those files.
